configure.ac

```
@@ -4171,33 +4171,42 @@ if test "$ac_cv_ffi_complex_double_supported" = "yes"; then
               [Defined if _Complex C type can be used with libffi.])
 fi
 
+# Check for use of the libmpdec library
+AC_MSG_CHECKING([for --with-libmpdec])
+AC_ARG_WITH(
+  [libmpdec],
+  [AS_HELP_STRING(
+    [--with-libmpdec],
+    [build _decimal module (default is yes)]
+  )])
+AC_MSG_RESULT([$with_libmpdec])
+
 # Check for use of the system libmpdec library
 AC_MSG_CHECKING([for --with-system-libmpdec])
+AC_DEFUN([USE_BUNDLED_LIBMPDEC],
+         [LIBMPDEC_CFLAGS="-I\$(srcdir)/Modules/_decimal/libmpdec"
+          LIBMPDEC_LIBS="-lm \$(LIBMPDEC_A)"
+          LIBMPDEC_INTERNAL="\$(LIBMPDEC_HEADERS) \$(LIBMPDEC_A)"
+          have_mpdec=yes
+          with_system_libmpdec=no])
 AC_ARG_WITH(
   [system_libmpdec],
   [AS_HELP_STRING(
     [--with-system-libmpdec],
     [build _decimal module using an installed mpdecimal library, see Doc/library/decimal.rst (default is yes)]
   )],
-  [],
+  [AS_IF([test "x$with_system_libmpdec" = xno && test "x$with_libmpdec" = xyes],
```

```suggestion
  [AS_IF([test "x$with_system_libmpdec" = xno -a "x$with_libmpdec" = xyes],
```

I think it's a bit more efficient that way and shorter

---

IIRC, ``-a`` is bashism: https://pubs.opengroup.org/onlinepubs/9799919799/

---

Wait, I thought it was POSIX! I may have introduced non-portable code then